src/main/java/com/redhat/devtools/lsp4ij/LSPIJUtils.java

```diff
@@ -595,12 +597,26 @@ public static int toOffset(int line, int character, @NotNull Document document)
 
         // Adjust position line/character according to this comment https://github.com/microsoft/vscode-languageserver-node/blob/ed3cd0f78c1495913bda7318ace2be7f968008af/textDocument/src/main.ts#L26
         if (line >= document.getLineCount()) {
-            // The line number is greater than the number of lines in a document, it defaults back to the number of lines in the document.
+            // The line number is greater than the number of lines in a document, it defaults to the last valid offset in the document.
             return document.getTextLength();
         } else if (line < 0) {
             // The line number is negative, it defaults to 0.
             return 0;
         }
+        // If the line number is 0 and the character is beyond the length of the first line, see if the language server
```

Comment:
The new logic is only engaged if the provided `Position` has `line=0` and `character>lines[0].length`, and only then if the client config feature flag `serverReportsOffsetOnlyPositions` is enabled. Otherwise it is the exact same logic as before that was ported from VS Code.